aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/AwsMetricAttributeGenerator.java

```diff
@@ -152,35 +212,132 @@ private static void setEgressOperation(SpanData span, AttributesBuilder builder)
-      logUnknownAttribute(AWS_REMOTE_OPERATION, span);
-      builder.put(AWS_REMOTE_OPERATION, UNKNOWN_REMOTE_OPERATION);
+      remoteService = GRAPHQL;
+      remoteOperation = getRemoteOperation(span, GRAPHQL_OPERATION_TYPE);
     }
 
     // Peer service takes priority as RemoteService over everything but AWS Remote.
     if (isKeyPresent(span, PEER_SERVICE) && !isKeyPresent(span, AWS_REMOTE_SERVICE)) {
-      setRemoteService(span, builder, PEER_SERVICE);
+      remoteService = getRemoteService(span, PEER_SERVICE);
     }
+
+    // try to derive RemoteService and RemoteOperation from the other un-directive attributes
+    if (remoteService.equals(UNKNOWN_REMOTE_SERVICE)) {
+      remoteService = generateRemoteService(span);
+    }
+    if (remoteOperation.equals(UNKNOWN_REMOTE_OPERATION)) {
+      remoteOperation = generateRemoteOperation(span);
+    }
+
+    builder.put(AWS_REMOTE_SERVICE, remoteService);
+    builder.put(AWS_REMOTE_OPERATION, remoteOperation);
+  }
+
+  /**
+   * When span name is not meaningful(null, unknown or http_method value) as operation name for http
+   * use cases. Will try to extract the operation name from http target string
+   */
+  private static String generateIngressOperation(SpanData span) {
+    String operation = UNKNOWN_OPERATION;
+    if (isKeyPresent(span, HTTP_TARGET)) {
+      String httpTarget = span.getAttributes().get(HTTP_TARGET);
+      // get the first part from API path string as operation value
+      // the more levels/parts we get from API path the higher chance for getting high cardinality
+      // data
+      if (httpTarget != null) {
+        operation = extractAPIPathValue(httpTarget);
+      }
+      if (isKeyPresent(span, HTTP_METHOD)) {
+        String httpMethod = span.getAttributes().get(HTTP_METHOD);
+        if (httpMethod != null) {
+          operation = httpMethod + " " + operation;
+        }
+      }
+    }
+    return operation;
+  }
+
+  /**
+   * When the remote call operation is undetermined for http use cases, will try to extract the
+   * remote operation name from http url string
+   */
+  private static String generateRemoteOperation(SpanData span) {
+    String remoteOperation = UNKNOWN_REMOTE_OPERATION;
+    if (isKeyPresent(span, HTTP_URL)) {
+      String httpUrl = span.getAttributes().get(HTTP_URL);
+      try {
+        URL url;
+        if (httpUrl != null) {
+          url = new URL(httpUrl);
+          remoteOperation = extractAPIPathValue(url.getPath());
+        }
+      } catch (MalformedURLException e) {
+        logger.log(Level.FINEST, "invalid http.url attribute: ", httpUrl);
+      }
+    }
+    if (isKeyPresent(span, HTTP_METHOD)) {
+      String httpMethod = span.getAttributes().get(HTTP_METHOD);
+      remoteOperation = httpMethod + " " + remoteOperation;
+    }
+    if (remoteOperation.equals(UNKNOWN_REMOTE_OPERATION)) {
+      logUnknownAttribute(AWS_REMOTE_OPERATION, span);
+    }
```

Comment:
Not blocking: This doesn't seem appropriate for this method - I would probably keep it in setRemoteServiceAndOperation. Ditto to generateRemoteService